lib/money/src/Money.cpp

```diff
@@ -0,0 +1,85 @@
+#include "Money.h"
+
+Money::Money(double amount, Currency currency)
+	: USCAmount(converter.ConvertToUSC(amount, currency))
+	, currency(currency)
+{}
+
+Money Money::operator+=(const Money other) {
+	this->USCAmount += other.USCAmount;
+	return *this;
+}
+
+Money Money::operator-=(const Money other) {
+	this->USCAmount -= other.USCAmount;
+	return *this;
+}
+
+std::ostream& operator<<(std::ostream& os, const Money other) {
+	return other.Print(os);
+}
+
+std::istream& operator>>(std::istream& is, Money& other) {
+	std::cout << "Input currency (USD, DKK): ";
+	std::string currency; 
+	std::cin >> currency;
+	if (!(currency == "USD" || currency == "DKK"))
+		throw Money::IncorrectInput{};
+	
+	std::cout << "Input amount: ";
+	double amount; 
+	if (!(std::cin >> amount)) 
+		throw Money::IncorrectInput{};
+
+	if (currency == "USD") {
+		other.currency = Currency::USD;
+		other.USCAmount = other.converter.ConvertToUSC(
+			amount,
+			other.currency
+		);
+	}
+	if (currency == "DKK") {
+		other.currency = Currency::DKK;
+		other.USCAmount = other.converter.ConvertToUSC(
+			amount,
+			other.currency
+		);
+	}
+	return is;
+}
+
+bool operator==(const Money lhs, const Money rhs) {
+	if (lhs.USCAmount == lhs.USCAmount &&
+		lhs.currency == lhs.currency)
```

Comment:
You compare the same object (lhs)
Avoid using such logic, just
 return lhs.amount == rhs.amount && lhs.currency == rhs.currency